src/TRestRawDreamToSignalProcess.cxx

```diff
@@ -0,0 +1,587 @@
+                                "number, isample = "
+                             << Feu.isample << " prev_isample = " << Feu.isample_prev << endl;
+                        bad_event = true;
+                    }
+                }
+                debug << "ReadFeuHeaders: header FeuHeaderLine " << Feu.FeuHeaderLine << " FineTimeStamp "
+                      << Feu.FineTimeStamp << " isample " << Feu.isample << endl;
+
+                // Reading optionals
+            } else if (Feu.FeuHeaderLine == 4) {
+                Feu.EventID_Op = Feu.current_data.get_data();
+                Feu.EventID += (1 << 12) * Feu.EventID_Op;
+                debug << "ReadFeuHeaders: header FeuHeaderLine " << Feu.FeuHeaderLine << " EventID_Op "
+                      << Feu.EventID_Op << endl;
+
+            } else if (Feu.FeuHeaderLine == 5) {
+                Feu.TimeStamp_Op1 = Feu.current_data.get_data();
+                debug << "ReadFeuHeaders: header FeuHeaderLine " << Feu.FeuHeaderLine << " TimeStamp_Op1 "
+                      << Feu.TimeStamp_Op1 << endl;
+
+            } else if (Feu.FeuHeaderLine == 6) {
+                Feu.TimeStamp_Op2 = Feu.current_data.get_data();
+                debug << "ReadFeuHeaders: header FeuHeaderLine " << Feu.FeuHeaderLine << " TimeStamp_Op2 "
+                      << Feu.TimeStamp_Op2 << endl;
+
+            } else if (Feu.FeuHeaderLine == 7) {
+                Feu.TimeStamp_Op3 = Feu.current_data.get_TimeStamp_Op();
+                Feu.TimeStamp += ((long long)1 << 36) * Feu.TimeStamp_Op3 + (1 << 24) * Feu.TimeStamp_Op2 +
+                                 (1 << 12) * Feu.TimeStamp_Op1;
+                debug << "ReadFeuHeaders: header FeuHeaderLine " << Feu.FeuHeaderLine << " TimeStamp_Op3 "
+                      << Feu.TimeStamp_Op3 << " TimeStamp " << Feu.TimeStamp << endl;
+            }
+
+            Feu.data_to_treat = false;
+            Feu.FeuHeaderLine++;
+
+        } else if (Feu.FeuHeaderLine > 8 && Feu.current_data.is_Feu_header()) {
+            ferr << "TRestRawDreamToSignalProcess::ReadFeuHeaders: too long Feu header part  "
+                 << Feu.FeuHeaderLine << endl;
+            bad_event = true;
+        } else if (Feu.FeuHeaderLine > 3 && !Feu.current_data.is_Feu_header())
+            break;  // header finished
+
+        if (fread((void*)&(Feu.current_data), sizeof(Feu.current_data), 1, fInputBinFile) == 0) return true;
+        totalBytesReaded += sizeof(Feu.current_data);
+        Feu.current_data.ntohs_();
+        Feu.data_to_treat = true;
+
+    }  // end while
+
+    fSignalEvent->SetID(Feu.EventID);
+    fSignalEvent->SetTime(tStart + Feu.TimeStamp * 8.E-9);  // timeStamp in seconds
+
+    return false;
+}
+
+bool TRestRawDreamToSignalProcess::ReadDreamData(FeuReadOut& Feu) {
+    register bool got_raw_data_header = false;
+    register bool got_channel_id = false;
+    register int ichannel = 0;
```

Comment:
`register` seems deplecated
https://stackoverflow.com/questions/3207018/register-keyword-in-c

Please consider to remove it

Reply:
So, what we should do? just remove it?

Reply:
I think so
`register bool got_raw_data_header` --> ` bool got_raw_data_header`